tests/sgx_zerobase/host/host.cpp

```diff
@@ -1,40 +1,54 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include <openenclave/bits/exception.h>
 #include <openenclave/host.h>
+#include <openenclave/internal/error.h>
 #include <openenclave/internal/tests.h>
+#include <sys/mman.h>
 #include <iostream>
-#include <vector>
+
+#include "../host/sgx/cpuid.h"
 #include "sgx_zerobase_u.h"
 
 const char* message = "Hello world from Host\n\0";
 
-int unsecure_string_patching(
-    const char* source,
-    char* output,
-    size_t output_length)
+static bool _is_misc_region_supported()
 {
-    size_t running_length = output_length;
-    while (running_length > 0 && *source != '\0')
-    {
-        *output = *source;
-        running_length--;
-        source++;
-        output++;
-    }
-    const char* ptr = message;
-    while (running_length > 0 && *ptr != '\0')
-    {
-        *output = *ptr;
-        running_length--;
-        ptr++;
-        output++;
-    }
-    if (running_length < 1)
-    {
+    uint32_t eax, ebx, ecx, edx;
+    eax = ebx = ecx = edx = 0;
+
+    // Obtain feature information using CPUID
+    oe_get_cpuid(CPUID_SGX_LEAF, 0x0, &eax, &ebx, &ecx, &edx);
+
+    // Check if EXINFO is supported by the processor
+    return (ebx & CPUID_SGX_MISC_EXINFO_MASK);
+}
+
+int test_ocall(const char* message)
+{
+    if (!message)
         return -1;
-    }
-    *output = '\0';
+
+    fprintf(stdout, "[host] Message from enclave : %s\n", message);
+
+    return 0;
+}
+
+int test_host_mmap(uint64_t addr)
+{
+    void* loc = mmap(
```

Comment:
Why do we need to test host mmap?

Reply:
I am actually not testing the mmap function itself, using it to test access to different memory addresses. I'll rename it to host_mmap().

Reply:
removed the function, since mmap is being used only once inside the host now.